packages/apollo-gateway/src/index.ts

```diff
@@ -143,20 +174,18 @@ export class ApolloGateway implements GraphQLService {
     config: GatewayConfig,
   ): Promise<[ServiceDefinition[], boolean]> {
     if (isLocalConfig(config)) return [config.localServiceList, false];
-    if (!config.serviceList)
-      throw new Error(
-        'The gateway requires a service list to be provided in the config',
-      );
-
-    const [
-      remoteServices,
-      isNewService,
-    ] = await getServiceDefinitionsFromRemoteEndpoint({
-      serviceList: config.serviceList,
-    });
 
-    this.createServices(remoteServices);
+    const [remoteServices, isNewService] = isConcreteConfig(config)
+      ? await getServiceDefinitionsFromRemoteEndpoint({
+          serviceList: config.serviceList,
+        })
+      : await getServiceDefinitionsFromStorage({
+          apiKey: config.apiKey || process.env['ENGINE_API_KEY']!,
+          graphVariant: config.tag || 'current',
+          federationVersion: config.federationVersion!,
+        });
 
+    this.createServices(remoteServices);
```

Comment:
iirc, @glasser pointed out that we do this in multiple places? Is that still the case here?

Reply:
You mean do we call `createServices` in multiple places? Yes:
1. We call it from `createSchema` after validating the services (`createSchema` is called on `load` for remote, or at construction for local)
2. We call it from `loadServiceDefinitions`, which is also called on `load`. 

So we call it twice on `load`. I pushed a new commit that removes "2".